### DISCUSSION TITLE
How to conveniently and quickly build a static curl

### DISCUSSION THREAD

**Discussion by cqcw123123** [curl/curl#15520] [General]
build a  static nginx is very simple：
$tar -zxf openssl-3.0.15.tar.gz
$tar -zxf pcre-8.45.tar.gz
$tar -zxf zlib-1.3.1.tar.gz
$tar -zxf nginx-1.27.2.tar.gz
$cd nginx-1.27.2
$ ./configure \
--with-cc-opt='-static -static-libgcc' \
--with-ld-opt=-static \
--prefix=/opt/nginx \
--with-http_ssl_module \
--with-http_v2_module \
--with-openssl=../openssl-3.0.15 \
--with-pcre=../pcre-8.45 \
--with-zlib=../zlib-1.3.1
$ make && make install

How to build a static curl?

**Comment by dfandrich** [comment]
Building a static curl is also very simple:

./configure --disable-shared --disable-ares --disable-ldap --disable-rtsp --without-ssl --without-zlib --without-brotli --without-zstd --without-libpsl --without-librtmp --without-libidn2 --without-nghttp2 --without-ngtcp2  CFLAGS='-static' LDFLAGS='-static'
make

Building a static curl with any features is considerably more difficult. See 
https://curl.se/docs/install.html#static-builds

**Reply by cqcw123123** [reply]
What I mean is that we need zlib, openssl, and so on

**Comment by bagder** [comment]
I recommend https://github.com/moparisthebest/static-curl/

**Reply by cqcw123123** [reply]
Yes, there is a static-curl  there, but I want to try compiling it myself, but their script can only run in Docker. It seems like I should learn Docker now.thank you.
﻿

**Comment by vszakats** [comment]
This script can also build it in one step: https://github.com/curl/curl-for-win
It supports Linux (arm64, amd64, riscv64), macOS and Windows.

**Reply by cqcw123123** [reply]
It looks complicated, so compiling nginx is very simple.

**Reply by vszakats** [reply]
Static linking is complicated, but it's basically similar to what nginx is automating in its custom `configure`: https://github.com/nginx/nginx/tree/master/auto
